pkg/openfeaturetest/testprovider.go

```diff
@@ -0,0 +1,108 @@
+package openfeaturetest
+
+import (
+	"context"
+	"fmt"
+	"github.com/open-feature/go-sdk/openfeature"
+	"runtime"
+	"sync"
+	"testing"
+)
+
+const testNameKey = "testName"
+
+// NewTestAwareProvider creates a new `TestAwareProvider`
+func NewTestAwareProvider() TestAwareProvider {
+	return TestAwareProvider{
+		providers: &sync.Map{},
+	}
+}
+
+// TestAwareProvider can be used in parallel unit tests. It holds a map of unit test name to `openfeature.FeatureProvider`s.
+// Before executing the test, specify the actual (in memory) provider that's going to  be used for the specific test using the
+// `SetProvider` method.
+type TestAwareProvider struct {
+	openfeature.NoopProvider
+	providers *sync.Map
```

Comment:
My only current issue with this approach is cleaning up this map. It would be great if we could have some ergonomic mechanism to REMOVE things from this map, eventually. Perhaps something a user could clean-up with a `defer` statement in whatever scope they need.